code/modules/reagents/reagent_containers.dm

```diff
@@ -15,20 +15,42 @@
 	var/temperature_min = 0 // To limit the temperature of a reagent container can atain when exposed to heat/cold
 	var/temperature_max = 10000
 
+/obj/item/reagent_containers/proc/can_set_transfer_amount(mob/user)
+	if(!length(possible_transfer_amounts))
+		// Nothing to configure.
+		return FALSE
+	if(isrobot(user) && src.loc == user)
+		// Borgs can configure their modules.
+		return TRUE
+	if(!Adjacent(user))
+		// No configuring the beaker across the room.
+		return FALSE
+	if(!ishuman(user))
+		// Although a funny idea, station pets changing transfer
+		// amounts on random conatiners would probably be frustrating
+		// for the crew.
+		return FALSE
+	if(HAS_TRAIT(user, TRAIT_HANDS_BLOCKED))
+		// I guess there's, like, a switch or a dial or something?
+		// Whatever, you need to use your hands for this.
+		return FALSE
+	return TRUE
+
 /obj/item/reagent_containers/AltClick(mob/user)
-	if(!Adjacent(user) || !length(possible_transfer_amounts) || !ishuman(user) || HAS_TRAIT(user, TRAIT_HANDS_BLOCKED))
+	if(!can_set_transfer_amount(user))
 		return
 
 	var/new_transfer_rate = tgui_input_list(user, "Amount per transfer from this:", "[src]", possible_transfer_amounts, "[amount_per_transfer_from_this]")
 	if(!new_transfer_rate)
 		return
 
-	if(!Adjacent(user))
-		to_chat(user, "<span class='warning'>You have moved too far away!</span>")
-		return
-	if(!ishuman(user) || HAS_TRAIT(user, TRAIT_HANDS_BLOCKED))
-		to_chat(user, "<span class='warning'>You can't use your hands!</span>")
-		return
+	if(!can_set_transfer_amount(user))
+		if(!Adjacent(user))
```

Comment:
This is already checked earlier in `can_set_transfer_amount` so it won't be reached. If you want the message to remain, then you can add the message to the check itself but add a param saying if to_chats should be shown

Reply:
Look again, please. As the new comment indicates, the second `can_set_transfer_amount` call is not redundant, because time elapsed with the input list open. The messages are very much reachable, as I saw them both in testing.

Reply:
Ah I'm blind yeah. I missed the `!` before the `can_set...`. This is fine as is then aye. Doing some checks double but that's fine for user experience